packages/serverless/src/logger.js

```diff
@@ -62,45 +62,15 @@ class InstanaServerlessLogger {
     return minLevel === DEBUG_LEVEL;
   }
 
-  info() {
-    if (!this.logger.info) {
-      return;
-    }
-
-    this.logger.info.apply(this.logger, arguments);
-  }
-
-  warn() {
-    if (!this.logger.warn) {
-      return;
-    }
+  warn = (...args) => this.logger?.warn?.(...args);
```

Comment:
See https://github.com/instana/nodejs/blob/v4.11.1/packages/aws-lambda/src/wrapper.js#L373-L392

When copying a fn for a class, the class itself needs to bind the this context (or the caller).

> logFn = logger.warn

Using the arrow fn here auto-resolves it. It allows to copy the fn to a new variable without lossing the binding.

Reply:
I will create a new PR for the collector logger soon. Its using a similar logic.